src/components/tables/RowActions/Delete/Button.tsx

```diff
@@ -0,0 +1,83 @@
+import { Button, Dialog } from '@mui/material';
+import DeleteConfirmation from 'components/tables/RowActions/Delete/Confirmation';
+import ProgressDialog from 'components/tables/RowActions/Delete/ProgressDialog';
+import {
+    SelectableTableStore,
+    selectableTableStoreSelectors,
+} from 'components/tables/Store';
+import { useConfirmationModalContext } from 'context/Confirmation';
+import { useZustandStore } from 'hooks/useZustand';
+import { useState } from 'react';
+import { FormattedMessage } from 'react-intl';
+
+function DeleteButton() {
+    const confirmationModalContext = useConfirmationModalContext();
+
+    const [showProgress, setShowProgress] = useState(false);
+    const [deleting, setDeleting] = useState<any[]>([]);
+
+    const selectedRows = useZustandStore<
+        SelectableTableStore,
+        SelectableTableStore['selected']
+    >(selectableTableStoreSelectors.selected.get);
+
+    const setAllSelected = useZustandStore<
+        SelectableTableStore,
+        SelectableTableStore['setAllSelected']
+    >(selectableTableStoreSelectors.selected.setAll);
+
+    const rows = useZustandStore<
+        SelectableTableStore,
+        SelectableTableStore['rows']
+    >(selectableTableStoreSelectors.rows.get);
+
+    const hasSelections = selectedRows.size > 0;
+
+    const handlers = {
+        delete: () => {
+            if (hasSelections) {
+                const deleteList: string[] = [];
```

Comment:
The terms used within the delete components are vague. In the interest of time, this can be revisited post-merge and I will stop commenting on it from this point forward. As aforementioned, my review began from the bottom up.

Reply:
That was kinda the point as I knew I was going to base the "shared" version of this off this.